src/Core/src/Platform/Android/MauiMaterialButton.cs

```diff
@@ -1,35 +1,169 @@
-﻿using Android.Content;
+﻿using System;
+using Android.Content;
+using Android.Graphics.Drawables;
+using Android.Views;
 using AndroidX.Core.Widget;
 using Google.Android.Material.Button;
 
 namespace Microsoft.Maui.Platform
 {
 	public class MauiMaterialButton : MaterialButton
 	{
-		// Currently Material doesn't have any bottom gravity options
-		// so we just move the layout to the bottom using
-		// SetCompoundDrawablesRelative during Layout
-		internal const int IconGravityBottom = 9999;
-		public MauiMaterialButton(Context context) : base(context)
+		// The default MaterialButton currently does not have a concept of bottom
+		// gravity which we need for .NET MAUI.
+		// In order to get this feature, we have added a custom gravity option
+		// that serves as a flag to indicate that the icon should be placed at
+		// the bottom.
+		// The real gravity value is IconGravityTop in order to perform all the
+		// normal layout calculations. We then set ForceBottomIconGravity for our
+		// custom layout pass where we simply swap the icon from the top to the
+		// bottom using SetCompoundDrawablesRelative.
+		internal const int IconGravityBottom = 0x1000;
+
+		public MauiMaterialButton(Context context)
+			: base(context)
 		{
 		}
 
-		protected override void OnLayout(bool changed, int left, int top, int right, int bottom)
+		public override int IconGravity
```

Comment:
If we set an invalid/new value here, then the measure and layout passes will not work properly. Luckily, we can pretend that bottom is top, and then swap to the bottom at the last minute after the layout has completed.